poppy/fresnel.py

```diff
@@ -1066,16 +1067,27 @@ def add_optic(self, optic=None, distance=0.0 * u.m):
             Some optic
         distance : astropy.Quantity of dimension length
             separation distance of this optic relative to the prior optic in the system.
+        index : int
+            Index at which to insert the new optical element
+
         """
-        self.planes.append(optic)
-        self.distances.append(distance.to(u.m))
+        if index is None:
```

Comment:
I could have sworn we already had this `index` parameter addition in some other PR already. Am I misremembering? Or was that just in the Fraunhofer code? 

In any case this is clearly good to have.

Reply:
Yes, I believe this was merged previously.  I'm not sure why this commit was group in with this PR again.